evm_arithmetization/src/cpu/kernel/interpreter.rs

```diff
@@ -64,6 +66,22 @@ pub(crate) struct Interpreter<F: RichField> {
     pub(crate) clock: usize,
     /// Log of the maximal number of CPU cycles in one segment execution.
     max_cpu_len_log: Option<usize>,
+    /// Optional logs for transactions code.
+    pub(crate) struct_logs: Option<Vec<Option<Vec<ZeroStructLog>>>>,
```

Comment:
For the outside `Option`, I think it would be nice to keep it indeed, since we can use it as a way to indicate whether the checks should be carried out. So I'll just wrap the internal values into an `OptionalStructLog`.

Reply:
In Einars branch, I have used something like
```
pub struct TxStructLogs(pub Option<TxHash>, pub Vec<StructLog>);
```
and then you would have
```
Option<Vec<TxStructLogs>>,
```